src/librustdoc/clean/mod.rs

```diff
@@ -973,9 +973,10 @@ impl Clean<Item> for hir::ImplItem<'_> {
         let local_did = self.def_id.to_def_id();
         cx.with_param_env(local_did, |cx| {
             let inner = match self.kind {
-                hir::ImplItemKind::Const(ref ty, expr) => {
-                    AssocConstItem(ty.clean(cx), Some(print_const_expr(cx.tcx, expr)))
-                }
+                hir::ImplItemKind::Const(ref ty, expr) => ConstantItem(Constant {
+                    type_: ty.clean(cx),
+                    kind: ConstantKind::Local { def_id: local_did, body: expr },
```

Comment:
The issue with this way of doing it is that it doesn't differentiate between "normal" constants and associated ones, which is why I think you have the issue in the search index generation.